### PR TITLE
feat: update contributing guide

### LINKED ISSUE #287: Feature: Update "Introduction to Contributing"
### Type of feature

📝 Documentation

### Current behavior

There are a couple of things that we need to update in the [contributing guide](https://docs.opensauced.pizza/contributing/introduction-to-contributing/):

1. **Create New Branch: Compliance**

   We have a [compliance action](https://github.com/open-sauced/hot/blob/beta/.github/workflows/compliance.yml#L47) that runs to check protected branch. If changes are made directly on default branch, it will give below message:

   ```yml
   ## Protected Branch

   In order to be considered for merging, the pull request changes must
   not be implemented on the "%branch%" branch. This is described in our
   [Contributing Guide](https://docs.opensauced.pizza/contributing/introduction-to-contributing/).
   We would suggest that you close this PR and implement your changes as
   described in our Contributing Guide and open a new pull request.
   ```

   Contributors are suggested to close the PR and create a new branch _as described in the contributing guide_. However, we don't have this stated anywhere in our contributing guide.

2. **Admonitions**

   Docusaurus supports special [admonitions](https://docusaurus.io/docs/markdown-features/admonitions) syntax to highlight note, tip, info, etc. There are some notes and tips that we can highlight in this page.

### Suggested solution

1. Update the instructions in [Pull Requests](https://docs.opensauced.pizza/contributing/introduction-to-contributing/#pull-requests-pr) section by emphasizing the importance of creating new branch and that PRs that has been made directly on default branch will be closed.
2. Add special Docusaurus special admonitions syntax for notes and tips accross the page.

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by adiati98**
## Description

This PR updates contributing guide with the following changes:

- Rewording an instruction to emphasize that creating a new branch is a must.
- Add an information that maintainers may unassign contributors and close their PR if they take more than one `good first issue`.
- Add Docusaurus special admonition syntax to highlight notes, infos, etc.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [x] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes #287 

### Before

Before applying admonition syntax

### After

Admonition highlight

## Steps to QA

1. Pull this branch.
2. Run the project locally.
3. Go to http://localhost:3000/contributing/introduction-to-contributing/

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [ ] Tier 4

### CODE REVIEW

**Comment by adiati98** [comment]
@BekahHW I have a note here.

The syntax for the "danger" one is actually `warning` (line 87). But it somehow renders as danger. It should be rendered as warning like on the [Docusaurus docs](https://docusaurus.io/docs/markdown-features/admonitions).
 
Hopefully, updating the Docusaurus in the future will resolve this. I included screenshots of before and after in the PR form. So, if you want me to change the "danger" to what we had previously, please let me know.

Thanks!

File: docs/contributing/introduction-to-contributing.md [hunk 1]
```diff
@@ -80,19 +81,31 @@ We actively welcome your pull requests. However, you must ensure that **you are
...
+:::info
 
-> **IMPORTANT:** <br />
-> Please do not DM Maintainers or contact them outside of GitHub if you want a PR Review. If you haven't received a review within a week, please tag them in the PR comments to ask for an estimated time of review.
+**IMPORTANT** <br /> Please do not DM Maintainers or contact them outside of GitHub if you want a PR Review. If you haven't received a review within a week, please tag them in the PR comments to ask for an estimated time of review.
 
-> _**Note for maintainers**: All PRs need a label to assist automation. See [this guide](./triage-guide.md#adding-issue-and-pr-labels) to determine which labels to use._
+:::
+
+:::note
```
Comment (BekahHW) [inline docs/contributing/introduction-to-contributing.md:104]: I think the back to back "special" blocks might be too much. Let's leave the note as is, esp bc it's more for maintainers and not necessarily something contributors really need.

File: docs/contributing/introduction-to-contributing.md [hunk 2]
```diff
@@ -80,19 +81,31 @@ We actively welcome your pull requests. However, you must ensure that **you are
...
+
+:::
 
-- the issue is not assigned to the contributor who opened the PR  
-- no issue is linked to the PR  
+:::info
 
-> **IMPORTANT:** <br />
-> Please do not DM Maintainers or contact them outside of GitHub if you want a PR Review. If you haven't received a review within a week, please tag them in the PR comments to ask for an estimated time of review.
+**IMPORTANT** <br /> Please do not DM Maintainers or contact them outside of GitHub if you want a PR Review. If you haven't received a review within a week, please tag them in the PR comments to ask for an estimated time of review.
```
Comment (BekahHW) [inline docs/contributing/introduction-to-contributing.md:100]: Let's move this to the community section right below where we mention joining Discord.

**Review by BekahHW — CHANGES_REQUESTED** [review]
Just a couple of things to break up the special notes.

**Comment by adiati98** [comment]
@BekahHW I made the changes here based on your suggestions.
One note, I moved the note for maintainer as part of point 9 as below screenshot because it talks about PR form. So, it make more sense to have this note as part of this point. But let me know if you want me to move it back to after the warning (danger) highlight. :)

note for maintainers

**Review by BekahHW — APPROVED** [review]

**Pull request merged**